File: src/main/java/net/hectus/neobb/turn/person_game/block/PTGlowstone.java
package net.hectus.neobb.turn.person_game.block;

import com.marcpg.libpg.util.Randomizer;
import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import net.hectus.neobb.turn.person_game.categorization.UtilityCategory;
import org.bukkit.block.Block;

public class PTGlowstone extends BlockTurn implements UtilityCategory {
    public PTGlowstone(NeoPlayer player) { super(player); }
    public PTGlowstone(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public void apply() {
        super.apply();
        new Buff.ExtraTurn(Buff.BuffTarget.YOU, Randomizer.boolByChance(25) ? 2 : 1).apply(player);
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/person_game/block/PTGlowstone.kt
package net.hectus.neobb.modes.turn.person_game.block

import com.marcpg.libpg.storing.Cord
import com.marcpg.libpg.util.Randomizer
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.modes.turn.default_game.block.BlockTurn
import net.hectus.neobb.modes.turn.person_game.categorization.UtilityCategory
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class PTGlowstone(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), UtilityCategory {
    override val damage: Double = 2.0

    override fun apply() {
        ExtraTurn(if (Randomizer.boolByChance(25.0)) 2 else 1).apply(player!!)
    }
}
